engine/src/main/java/com/datatorrent/stram/util/LoggerUtil.java

```diff
@@ -355,4 +366,91 @@ private static boolean isErrorLevelEnable()
     return false;
   }
 
+  /**
+   * Adds Logger Appender
+   * @param name Appender name
+   * @param refName Appender reference name (appender name in the properties)
+   * @param properties Appender properties
+   */
+  public static void addAppender(String name, String refName, Properties properties)
+  {
+    if (getAppendersNames().contains(name)) {
+      logger.warn("A logger appender with the name '{}' exists. Cannot add a new logger appender with the same name", name);
+    } else {
+      try {
+        Method method = PropertyConfigurator.class.getDeclaredMethod("parseAppender", Properties.class, String.class);
+        method.setAccessible(true);
+        Appender appender = (Appender)method.invoke(new PropertyConfigurator(), properties, refName);
+        if (appender == null) {
+          logger.warn("Cannot add a new logger appender. Name: {}, Reference name: {}, Properties: {}", name, refName, properties);
+        } else {
+          appender.setName(name);
```

Comment:
See if you can avoid overriding the appender name and let the original refname specified by the user remain.